en/middleware/xrce_dds.md

```diff
@@ -0,0 +1,303 @@
+:::
+
+To start the agent with settings for connecting to the XRCE-DDS client running on the simulator:
+
+```sh
+MicroXRCEAgent udp4 -p 8888
+```
+
+### Install from Snap Package
+
+Install from a snap package on Ubuntu using the following command:
+
+```sh
+sudo snap install micro-xrce-dds-agent --edge
+```
+
+To start the agent with settings for connecting to the XRCE-DDS client running on the simulator (note that the command name is different than if you build the agent locally):
+
+```sh
+micro-xrce-dds-agent udp4 -p 8888
+```
+
+:::note
+At time of writing the stable of version installed from snap connects to PX4 but reports errors creating topics.
+The development version, fetched using `--edge` above, does work.
+:::
+
+### Build/Run within ROS2 Workspace
+
+The agent can be built and launched within a ROS2 workspace (or build standalone and launched from a workspace.
+You must already have installed ROS 2 following the instructions in: [ROS 2 User Guide > Install ROS 2](##TBDXXX)
+
+To build the agent within ROS:
+
+1. Create a workspace directory for the agent:
+
+   ```sh
+   mkdir -p ~/ws_xrce_dds_agent/src
+   ```
+
+1. Clone the source code for the eProsima [Micro-XRCE-DDS-Agent](https://github.com/eProsima/Micro-XRCE-DDS-Agent) to the `/src` directory (the `main` branch is cloned by default):
+
+   ```sh
+   cd ~/px4_ros_xrce_dds_ws/src
+   git clone https://github.com/eProsima/Micro-XRCE-DDS-Agent.git
+   ```
+1. Source the ROS2 development environment, in this case "ROS 2 foxy", and compile the workspace using `colcon`:
+
+   ```sh
+   cd ..
+   source /opt/ros/foxy/setup.bash
+   colcon build
+   ```
+   This builds all the folders under `/src` using the sourced toolchain.
+
+
+To run the XRCE-DDS agent in the workspace:
+
+1. Source the `local_setup.bash` to make the executables available in the terminal (also `setup.bash` if using a new terminal).
+
+   ```sh
+   source /opt/ros/foxy/setup.bash
+   source install/local_setup.bash
+   ```
+1. Start the agent with settings for connecting to the XRCE-DDS client running on the simulator: 
+
+   ```sh
+   MicroXRCEAgent udp4 -p 8888
+   ```
+
+## Starting XRCE-DDS
+
+### Starting the Agent
+
+The agent is used to connect to the client over a particular channel, such as UDP, TCP, or a serial connection.
+The channel settings are specified when the agent is started, using command line options. 
+These are documented in the eProsima user guide: [Micro XRCE-DDS Agent > Agent CLI](https://micro-xrce-dds.docs.eprosima.com/en/latest/agent.html#agent-cli). <!-- what options work? i.e. does PX4 allow CAN? -->
+
+:::note
+You should create a single instance of the agent for each channel over which you need to connect.
+:::
+
+For example, the PX4 simulator runs the XRCE-DDS client over UDP on port 8888, so to connect to the simulator you would start the agent with the command:
+
+```sh
+MicroXRCEAgent udp4 -p 8888
+```
+
+### Starting the Client
```

Comment:
There is one last method to start the client on real vehicles: using the parameter `XRCE_DDS_0_CFG` of the group [Micro XRCE-DDS](https://docs.px4.io/main/en/advanced_config/parameter_reference.html#micro-xrce-dds). With that you can start the client on serial or Ethernet or even WiFi (it seems). Then you have other parameters, like `XRCE_DDS_UDP_PRT` to setup the UDP port, `XRCE_DDS_DOM_ID` if you want to connect to a different `DOMAIN_ID` (https://docs.ros.org/en/foxy/Concepts/About-Domain-ID.html) and finally `XRCE_DDS_KEY` if you want to change the default `SESSION_ID` (needed for multi vehicle simulations). Anyway, I think the last two parameters can be put in the docs later on.

Reply:
@beniaminopozzan Thanks! Actually you should ALWAYS use the parameters on hardware. If you set the command you're setting yourself up for conflicts with other configuration. I have mentioned it still, but not as recommended.

I have rewritten this section, can you check?

Reply:
Thanks!!!

For serial connection I think it is perfect, for Ethernet connection it seems that @ozitoune managed to have it working on Pixhawk 6X with
```
microdds_client start -t udp -p 8888 -h 192.168.0.100
```
where `-h 192.168.0.100` tells the IP address of the _Agent_. As this is not a parameter in PX4, it is then required to start the client manually or from System Startup.

Reply:
@beniaminopozzan That's horrible - can you create an issue to get a parameter for the host into PX4?